cupy/__init__.py

```diff
@@ -341,12 +341,39 @@ def binary_repr(num, width=None):
 # -----------------------------------------------------------------------------
 # Data type routines (borrowed from NumPy)
 # -----------------------------------------------------------------------------
-from numpy import can_cast  # NOQA
-from numpy import common_type  # NOQA
+def can_cast(from_, to, casting='safe'):
+    """Returns True if cast between data types can occur according to the
+    casting rule. If from is a scalar or array scalar, also returns True if the
+    scalar value can be cast without overflow or truncation to an integer.
+
+    .. seealso:: :func:`numpy.can_cast`
+    """
+    from_dtype = getattr(from_, 'dtype', from_)
+    return numpy.can_cast(from_dtype, to, casting=casting)
```

Comment:
This implementation seems wrong.
```py
>>> numpy.can_cast(numpy.int64(1), 'int16')
True
>>> cupy.can_cast(cupy.int64(1), 'int16')
False
```
How about implementing as follows?
```py
from_dtype = from_.dtype if isinstance(from_, cupy.ndarray) else from_
return numpy.can_cast(from_dtype, to, casting=casting)
```

Reply:
The variable name `from_dtype` seems a little confusing. How is `from_no_cupy`, for instance?

Reply:
Good catch, I will make the changes.

Reply:
Regarding confusing variable names, to avoid that, I simply overwrote the original `from_`, this way it leaves less room for what the variable means.